### DISCUSSION TITLE
Curl Issue Comment File Attachment (Private Repository)

### DISCUSSION THREAD

**Discussion by FWBlueWizard** [community/community#36719] [Projects and Issues]
I have two repositories one private and one public. I am using some workflows that take a user created file that is posted as an attachment to a comment and using it as part of the workflow data. 

I don't have any issue on the public repository to `curl -L -O -sS https://github.com/<owner>/<repo>/files/<some number>/<filename>`

What is the reason this same call does not work with a private repository? 
My first assumption was maybe I need to authenticate so now provide the header 

curl -L -O -sS -H "Authorization: Bearer <TOKEN>"  https://github.com/<owner>/<repo>/files/<some number>/<filename>
From what I can tell authentication is good 
```bash
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=github.com
*  start date: Mar 15 00:00:00 2022 GMT
*  expire date: Mar 15 23:59:59 2023 GMT
*  subjectAltName: host "github.com" matched cert's "github.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert TLS Hybrid ECC SHA384 2020 CA1
*  SSL certificate verify ok.
```
But 
I get a 404 error and the file that is downloaded is an ASCII file with contents "Not Found" 

I'm really at a loss... Files are attached like this Test.txt file in this comment. What am I doing wrong? Do I need a different HEADER? an additioanl header with "Accept: ? " which data type does it need to be. 

Thanks for any help! 

[Test.txt](https://github.com/community/community/files/9824945/Test.txt)

**Comment by ossanna16** [comment]
Hey there @F-WRunTime and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by FWBlueWizard** [comment]
This is an issue for us. How else does a user provide input 'files' and not just 'input' arguments from a comment?

**Comment by chubin** [comment]
Have you solved this issue? I have the same problem and I have found some workarounds for the moment, but I don't really like them, and I am searching for a real solution (I am not sure that it is possible though)
